pkg/kubelet/eviction/threshold_notifier_linux.go

```diff
@@ -67,51 +70,118 @@ func NewMemCGThresholdNotifier(path, attribute, threshold, description string, h
 			unix.Close(eventfd)
 		}
 	}()
-	glog.V(3).Infof("eviction: setting notification threshold to %s", threshold)
-	config := fmt.Sprintf("%d %d %s", eventfd, watchfd, threshold)
+	epfd, err := unix.EpollCreate1(0)
+	if err != nil {
+		glog.Warningf("eviction manager: error creating epoll: %v", err)
+		return nil, err
+	}
+	if epfd < 0 {
+		err = fmt.Errorf("EpollCreate1 call failed")
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			unix.Close(epfd)
+		}
+	}()
+	config := fmt.Sprintf("%d %d %d", eventfd, watchfd, threshold)
 	_, err = unix.Write(controlfd, []byte(config))
```

Comment:
I'm fairly sure you don't need to keep controlfd around after you do this write, so you could just close it and not store the fd in &linuxCgroupNotifier...

I'm also like 99% sure you don't need the watchfd either, could be closed here too, no need to store it.

Reply:
nice catch.  Removed (and tested to make sure it is safe)